packages/web-frontend/src/selectors/measureSelectors.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * Tupaia Web
+ * Copyright (c) 2020 Beyond Essential Systems Pty Ltd.
+ * This source code is licensed under the AGPL-3.0 license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { createSelector } from 'reselect';
+import createCachedSelector from 're-reselect';
+
+import { getMeasureFromHierarchy } from '../utils';
+import {
+  getMeasureDisplayInfo,
+  POLYGON_MEASURE_TYPES,
+  calculateRadiusScaleFactor,
+} from '../utils/measures';
+import { getLocationComponentValue, URL_COMPONENTS } from '../historyNavigation';
+import { safeGet, selectLocation, getOrgUnitFromCountry } from './utils';
+import {
+  selectCurrentOrgUnitCode,
+  selectAncestors,
+  selectActiveProjectCountries,
+  selectCountryHierarchy,
+  selectAllOrgUnitsInCountry,
+  selectDescendantsFromCache,
+} from './orgUnitSelectors';
+import { selectCurrentProjectCode } from './projectSelectors';
+
+const displayInfoCache = createCachedSelector(
+  [
+    measureOptions => measureOptions,
+    (_, hiddenMeasures) => hiddenMeasures,
+    (_, __, data) => data,
+    (_, __, ___, organisationUnitCode) => organisationUnitCode,
+  ],
+  (options = [], hiddenMeasures, data, organisationUnitCode) => {
+    return {
+      organisationUnitCode,
+      ...data,
+      ...getMeasureDisplayInfo({ ...data }, options, hiddenMeasures),
+    };
+  },
+)((_, __, ___, organisationUnitCode) => organisationUnitCode);
+
+const getOrgUnitFromMeasureData = (measureData, code) =>
+  measureData.find(val => val.organisationUnitCode === code);
+
+// TODO: Refactor to be more general
```

Comment:
Should we address this item? No need to, just checking

Reply:
I have removed the TODO. I don't really think it is necessary.